cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java

```diff
@@ -373,12 +388,75 @@ private void stopWorkflowAdapter(Id.Namespace namespace, AdapterSpecification ad
     store.deleteSchedule(workflowId, SchedulableProgramType.WORKFLOW, scheduleName);
   }
 
-  private void startWorkerAdapter(Id.Namespace namespace, AdapterSpecification adapterSpec) {
-    // TODO: implement
+  private Id.Program getWorkerId(Id.Namespace namespace, AdapterSpecification adapterSpec) throws NotFoundException {
+    Id.Application appId = Id.Application.from(namespace, adapterSpec.getTemplate());
+    ApplicationSpecification appSpec = store.getApplication(appId);
+    if (appSpec == null || appSpec.getWorkers().size() != 1) {
```

Comment:
these 2 errors are different and should be handled differently i think.  though how can the second case happen?  It seems like if it happens it's a bug in our code?